### ISSUE TITLE
MAC Agent gets unhealthy on adding Elastic Defend integration and endpoint service fails to install.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/elastic-agent#4302]
**Kibana Build details:**
```
VERSION: 8.13.0
BUILD: 71738
COMMIT: b036a9705a55f6c81d065011ad8c991cbc3101d9
Artifact Link: https://staging.elastic.co/8.13.0-4304292e/downloads/beats/elastic-agent/elastic-agent-8.13.0-darwin-x86_64.tar.gz
```

**Host OS:**  MAC

**Preconditions:**
1. 8.13.0-BC1 Kibana cloud environment should be available.

**Steps to reproduce:**
1. Install MAC agent with Elastic Defend.
2. Observe Agent gets unhealthy with Elastic Defend integration and endpoint service fails to install in logs.

**Screenshot:**
![image](https://github.com/elastic/elastic-agent/assets/77374876/02a96175-5b7a-4d52-95b2-89ff81c9d6b2)

![image](https://github.com/elastic/elastic-agent/assets/77374876/b3d35fc1-059e-4c88-89d3-7877eaeb3505)

**Expected Result:**
Agent should remain healthy on adding Elastic Defend integration and should install endpoint service successfully.

**Logs:**
[elastic-agent-diagnostics-2024-02-21T05-58-10Z-00.zip](https://github.com/elastic/elastic-agent/files/14354694/elastic-agent-diagnostics-2024-02-21T05-58-10Z-00.zip)

**What's working fine:**
- No issue is observed on Windows and Linux agent with Elastic Defend integration.
![image](https://github.com/elastic/elastic-agent/assets/77374876/b45ce714-5fae-46b9-9a3d-a20094ad6352)

**Comment by amolnater-qasource** [comment]
@manishgupta-qasource Please review.

**Comment by manishgupta-qasource** [comment]
Secondary review for this ticket is Done

**Comment by cmacknz** [comment]
I see we fail running the endpoint command:

```json
{"log.level":"info","@timestamp":"2024-02-21T05:38:28.866Z","log.logger":"component.runtime.endpoint-default.service_runtime","log.origin":{"file.name":"runtime/service.go","file.line":315},"message":"after check if endpoint service is installed, err: exit status 135","ecs.version":"1.6.0"}
```

https://github.com/elastic/elastic-agent/blob/bdd885c9df3eb0c81624f852c74c64fc250a1b17/pkg/component/runtime/service.go#L314-L315

Where the `check` step is implemented as the `verify` command:

https://github.com/elastic/elastic-agent/blob/bdd885c9df3eb0c81624f852c74c64fc250a1b17/specs/endpoint-security.spec.yml#L27-L32

@nfritts what would cause `endpoint-security verify --log stderr` to run `exit status 135` on an X86 Mac immediately? There are no endpoint logs to look at here because we were never able to install it.

**Comment by nfritts** [comment]
I don't have time to look further, but I'm pretty sure 135 is:

```
    /// @brief Process is running in emulation on non-native CPU architecture
    Error_RunningInEmulation = -135,
```

Any chance its running the ARM binary for some reason?  or running the x86 binaries on an ARM mac?

**Comment by amolnater-qasource** [comment]
Hi @nfritts 

Thank you for looking into this.
Yes, we were installing x86_64 agent on ARM mac.

**Screenshot:**
![image](https://github.com/elastic/elastic-agent/assets/77374876/676289c0-2f2c-4380-8ad0-2b84c70dd5e5)

However, we never got any issues till 8.12.2 while installing x86_64 agent on ARM mac.
Please let us know, if this is expected now to not to install x86_64 on ARM mac .

**8.12.2-(Working fine):**
[elastic-agent-diagnostics-2024-03-04T09-33-23Z-00.zip](https://github.com/elastic/elastic-agent/files/14479433/elastic-agent-diagnostics-2024-03-04T09-33-23Z-00.zip)

**8.13.0 Logs-Issue:**
[elastic-agent-diagnostics-2024-03-04T06-29-10Z-00.zip](https://github.com/elastic/elastic-agent/files/14479415/elastic-agent-diagnostics-2024-03-04T06-29-10Z-00.zip)

Further, we confirm that with aarch64 artifact the issue is not reproducible and host is visible under Endpoints tab.

**Build details:**
VERSION: 8.13.0 BC3
BUILD: 71857
COMMIT: 82f46148c91eec93ac7382147936028db2eb8883
Artifact Link: https://staging.elastic.co/8.13.0-c16d8e3a/summary-8.13.0.html#elastic-agent-package


Thanks!

**Comment by nfritts** [comment]
We need to figure out if this is a bug or not.  

We added code to detect if we were running an x86 binary under ARM or not for 8.13 and updated the package spec to prevent doing so under Windows.

Mac is slightly different since we build a mac binary that can run on both and ship it in both versions of Agent.  This might be an unexpected "bug" that is part of that change, but honestly it might be a good thing.  running the x86 endpoint is going to be less performant on arm processors and we may want to avoid it.  

@ricardoungureanu what do you think?  do we want the endpoint to run in the native CPU architecture?

**Comment by amolnater-qasource** [comment]
Thank you for the confirmation @nfritts 

As MAC-ARM64 artifacts are working fine with ARM machine, we are closing this issue and marking as **QA:Validated**.

Thanks!!

**Comment by cmacknz** [comment]
Exit status 135 means endpoint is running on an emulated platform (for example x86 binary on an arm processor). Detection of this condition was added in 8.13, I don't believe endpoint supports this mode of operation any more.

CC @nfritts

**Comment by EchoGangster** [comment]
> Exit status 135 means endpoint is running on an emulated platform (for example x86 binary on an arm processor). Detection of this condition was added in 8.13, I don't believe endpoint supports this mode of operation any more.
> 
> 
> 
> CC @nfritts 

I installed the Darwin agent. I took a look at the elastic-agent.spec.yml file and it doesn't mention Apple in the file. Not sure if that is an issue. Might try and pull down an earlier version to see what the parameters are.

**Comment by cmacknz** [comment]
The usual cause for this error would be running the `darwin-x86` executable on an M1/M2/M3 ARM Mac or running the `darwin-aarch64` on an x86 executable.

**Comment by ZainULArfeen** [comment]
![image](https://github.com/elastic/elastic-agent/assets/37686241/9f323add-b016-4420-94c2-d0d7e6e43b7f)
![image](https://github.com/elastic/elastic-agent/assets/37686241/c0deccdf-8dec-4f47-ab4c-98d83e37370b)

i have the same problem is there any fix available for this or i have to downgrade to 8.12

**Comment by ZainULArfeen** [comment]
@EchoGangster thanks bud, it worked
